subsys/nrf_security/src/drivers/Kconfig

```diff
@@ -4,56 +4,229 @@
-		   (PSA_CRYPTO_DRIVER_CC3XX || PSA_CRYPTO_DRIVER_OBERON)
+	depends on PSA_WANT_ALG_ECDSA             || \
+			   PSA_WANT_ALG_RSA_PKCS1V15_SIGN || \
+			   PSA_WANT_ALG_RSA_PSS
+	help
+	  Enable CryptoCell for the PSA asymmetric signature APIs. Enabling this
+	  will provide support signing/verification with ECDSA in deterministic
+	  and randomized modes and RSA in PKCS1V15 and PSS modes.
 
+config PSA_USE_CC3XX_ASYMMETRIC_DRIVER
+	prompt "Enable CryptoCell for the PSA asymmetric encryption APIs"
+	bool
+	default y
+	depends on PSA_WANT_ALG_RSA_PKCS1V15_CRYPT || \
+			   PSA_WANT_ALG_RSA_OAEP
+	help
+	  Enable CryptoCell for the PSA asymmetric encryption APIs. Enabling this
+	  will provide support for assymetric encryption with RSA
+	  in PKCS1V15 and OAEP modes.
 
-rsource "nrf_cc3xx/Kconfig"
+config PSA_USE_CC3XX_KEY_PAIR_DRIVER
+	prompt "Enable CryptoCell for the PSA key management APIs"
+	bool
+	default y
+	depends on PSA_USE_CC3XX_ASYMMETRIC_DRIVER || \
+			   PSA_USE_CC3XX_SIGNATURE_DRIVER  || \
+			   PSA_USE_CC3XX_ECDH_DRIVER
+	help
+	  Enabling this will provide support for key generation, key importing and
+	  public exporting for asymmetric keys. ECC and RSA keys are supported.
+
+endif
+
+config PSA_USE_CC3XX_CTR_DRBG_DRIVER
+	bool
+	default y
+	depends on PSA_USE_CTR_DRBG_DRIVER
+	depends on NRF_CC3XX_PLATFORM || BUILD_WITH_TFM
+
+config PSA_USE_CC3XX_HMAC_DRBG_DRIVER
+	bool
+	default y
+	depends on PSA_USE_HMAC_DRBG_DRIVER
+	depends on NRF_CC3XX_PLATFORM || BUILD_WITH_TFM
+
+endmenu
+
+menu "AES key size configuration"
+
+config PSA_WANT_AES_KEY_SIZE_128
+	prompt "AES 128 bits key"
+	bool
 
-rsource "nrf_cc3xx_platform/Kconfig"
+config PSA_WANT_AES_KEY_SIZE_192
+	prompt "AES 192 bits key"
+	bool
+
+config PSA_WANT_AES_KEY_SIZE_256
+	prompt "AES 256 bits key"
+	bool
+
+endmenu
+
+menu "RSA key size configuration"
+
+config PSA_WANT_RSA_KEY_SIZE_1024
+	prompt "RSA 1024 bits key (unsafe)"
+	bool
+	default y
+	help
+	  RSA with 1024 bit keys are not recommended for new designs.
+	  Please see https://www.keylength.com/
+
+config PSA_WANT_RSA_KEY_SIZE_1536
+	prompt "RSA 1536 bits key (unsafe)"
+	bool
+	help
+	  RSA with 1536 bit keys are not recommended for new designs.
+	  Please see https://www.keylength.com/
+
+config PSA_WANT_RSA_KEY_SIZE_2048
+	prompt "RSA 2048 bits key"
+	bool
+
+config PSA_WANT_RSA_KEY_SIZE_3072
+	prompt "RSA 3072 bits key"
+	bool
+
+config PSA_WANT_RSA_KEY_SIZE_4096
+	prompt "RSA 4096 bits key"
+	bool
+
+config PSA_WANT_RSA_KEY_SIZE_6144
+	prompt "RSA 6144 bits key"
+	bool
+
+config PSA_WANT_RSA_KEY_SIZE_8192
+	prompt "RSA 8192 bits key"
+	bool
+
+config PSA_MAX_RSA_KEY_BITS
+	int
+	default 8192 if PSA_WANT_RSA_KEY_SIZE_8192
+	default 6144 if PSA_WANT_RSA_KEY_SIZE_6144
+	default 4096 if PSA_WANT_RSA_KEY_SIZE_4096
+	default 3072 if PSA_WANT_RSA_KEY_SIZE_3072
+	default 2048 if PSA_WANT_RSA_KEY_SIZE_2048
+	default 1536 if PSA_WANT_RSA_KEY_SIZE_1536
+	default 1024 if PSA_WANT_RSA_KEY_SIZE_1024
+	default 0
```

Comment:
This is a PSA-prefixed configuration that we made (that doesn't exist in PSA crypto APIs or Mbed TLS). Please add a help-text to note that this is non-standard configuration.

Reply:
Hm, the thing is that all the configurations in this file is custom from us. So I don't think that it makes sense to add the same help-text to every single configuration of this file.  What do you think about renaming the comment "PSA Driver support"  that is on top of all of these configurations to something like "Oberon PSA core configurations" or something similar?

Reply:
I'm confused, do the PSA_WANT_*_KEY_SIZE_* symbols exist as a concept in the Oberon PSA Core?

I thought it was something nordic had invented?

I agree with Frank that we need to somehow document whether symbols come from Nordic, Oberon, or ARM.

Reply:
The PSA_WANT_KEY_SIZE is an Oberon PSA core symbol indeed. And I do agree that we need to clarify it but not in every single symbol since it gets repetitive without much added value in my opinion. My proposal it to note that in the group so that we mention it once for all the configs.